src/peergos/server/space/JdbcQuotas.java

```diff
@@ -0,0 +1,149 @@
+package peergos.server.space;
+
+import peergos.server.sql.*;
+import peergos.server.util.Logging;
+
+import java.sql.*;
+import java.util.*;
+import java.util.function.*;
+import java.util.logging.*;
+
+public class JdbcQuotas {
+	private static final Logger LOG = Logging.LOG();
+
+    private static final String QUOTA_USER_NAME = "name";
+    private static final String QUOTA_SIZE = "quota";
+    private static final String SET_QUOTA = "UPDATE freequotas SET quota = ? WHERE name = ?;";
+    private static final String GET_QUOTA = "SELECT quota FROM freequotas WHERE name = ?;";
+    private static final String GET_ALL_QUOTAS = "SELECT name, quota FROM freequotas;";
+    private static final String REMOVE_USER = "DELETE FROM freequotas WHERE name = ?;";
+
+    private final SqlSupplier commands;
+    private volatile boolean isClosed;
+    private Supplier<Connection> conn;
+
+    public JdbcQuotas(Supplier<Connection> conn, SqlSupplier commands) {
+        this.commands = commands;
+        this.conn = conn;
+        init(commands);
+    }
+
+    private Connection getConnection() {
+        Connection connection = conn.get();
+        try {
+            connection.setAutoCommit(true);
+            connection.setTransactionIsolation(Connection.TRANSACTION_SERIALIZABLE);
+            return connection;
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private synchronized void init(SqlSupplier commands) {
+        if (isClosed)
+            return;
+
+        try (Connection conn = getConnection()) {
+            commands.createTable(commands.createQuotasTableCommand(), conn);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void setQuota(String username, long quota) {
+        try (Connection conn = getConnection();
+             PreparedStatement createuser = conn.prepareStatement(commands.insertOrIgnoreCommand("INSERT ", "INTO freequotas (name, quota) VALUES(?, ?)"));
```

Comment:
extract sql string to constant as per other queries

Reply:
It can't be a static constant because it's generated dynamically.

Reply:
I meant "INTO freequotas (name, quota) VALUES(?, ?)"

Reply:
But it doesn't make any sense on it's own